build_tools/scripts/byo_llvm.sh

```diff
@@ -175,6 +186,18 @@ do_build_iree() {
   cmake -DCMAKE_INSTALL_PREFIX="${iree_install_dir}" -P "${iree_build_dir}/cmake_install.cmake"
 }
 
+do_test_iree() {
+  echo "*********************** TESTING IREE **********************************"
+  iree_build_dir="${IREE_BYOLLVM_BUILD_DIR}/iree"
+  iree_install_dir="${IREE_BYOLLVM_INSTALL_DIR}/iree"
+
+  echo "Source Directory: ${REPO_ROOT}"
+  echo "Build Directory: ${iree_build_dir}"
+
+  cmake --build "${iree_build_dir}" --target iree-test-deps
+  "${REPO_ROOT}/build_tools/cmake/ctest_all.sh" "${iree_build_dir}"
```

Comment:
This feels similar to code I had concerns with at https://github.com/openxla/iree/pull/12921#discussion_r1157835074

I'm weary of any sort of meta-build scripting. If someone wants to run IREE tests _with some options_, they should run ctest directly. Putting convenience functions like this in scripts we recommend people use will tempt them to add additional passthrough variables or options, and then we're stuck maintaining two (or more) separate ways to run tests.

Is it really that important/useful to be able to run `build_tools/scripts/byo_llvm.sh test_iree` instead of using `cmake` and `ctest` directly? Those commands don't have anything "byo llvm"-specific in them.

Reply:
Summary: treat this `test_iree` action, which as you point out is implemented as just 2 commands so is nearly trivial, as documentation, and as an opportunity to advertise and align what is fully supported and tested on CI and what we teach users to do by default if they want to test their just-completed byo-llvm build.

I believe that it will be a common occurrence that users who have just completed a byo-llvm build will wonder if their custom, off-the-standard-path build is actually correct, and will want to reach for a "test what I just built" command, and even though /we/ know how easy it is to test that, some users will not know that. After all, it's not just a generic matter of "run `ctest`" since one needs to build `iree-test-deps` first, so this is something that many users will forget to do and get `ctest` failures which they will interpret as a defect in their byo-llvm build, and other users will find about that but only after painstaking docs reading and interpretation, since they will have to go read docs written around the normal bundled-LLVM path and they will have to interpret what applies to them and what does not. Again, none of that is a problem if the user is named Scott or Stella, but mankind is wonderfully diverse.

Reply:
FYI, we do have a `iree-run-tests` utility target that builds `iree-test-deps` first. That, and other ways to run tests, is documented at https://openxla.github.io/iree/building-from-source/getting-started/#running-tests.

I'm not _too_ concerned with the duplication/complexity here, but I do want to funnel developers down the already documented paths where possible.

Reply:
OK, dropped `test_iree`, inlined its 2 commands into `build_and_test_byo_llvm.sh`. 

Gave a try to `iree-run-tests` but was really missing features from `ctest_all.sh`. Would have had to manually set the `PARALLEL_LEVEL`, duplicating how `ctest_all.sh` does it. Then we would also potentially have missed the curated exclude-filters and knobs from `ctest_all.sh`.